pkg/watcher/watcher_test.go

```diff
@@ -0,0 +1,221 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Tetragon
+
+package watcher
+
+import (
+	"context"
+	"fmt"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/cilium/tetragon/pkg/logger"
+	"github.com/stretchr/testify/require"
+
+	"github.com/google/uuid"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	k8stypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes/fake"
+	k8sfake "k8s.io/client-go/kubernetes/fake"
+	clienttesting "k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/cache"
+)
+
+type tlog struct {
+	*testing.T
+	Logger *logrus.Logger
+}
+
+func (tl tlog) Write(p []byte) (n int, err error) {
+	tl.Log(string(p))
+	return len(p), nil
+}
+
+func TestFastK8s(t *testing.T) {
```

Comment:
nit: it's not super explicit what the test does from its name, maybe we can add some comments here from the commit message or change the name?

Reply:
Added a comment, thanks!